src/include/ntp-client/misc.rb

```diff
@@ -113,7 +111,7 @@ def string2opts(options_string, with_param, without_param)
           next { e => nil }
         end
       end
-      ret = Builtins.filter(ret) { |k, v| v != nil }
+      ret = Builtins.filter(ret) { |_k, v| !v.nil? }
```

Comment:
And also #select but if I change it here then i should clean up the whole method